### PR TITLE
Add shared type for all Quantities

### PR DESCRIPTION
**Pull request by ela-kotulska-frequenz**

### CODE REVIEW

File: RELEASE_NOTES.md [hunk 1]
```diff
@@ -2,8 +2,16 @@
 
 ## Summary
 
-This is the initial release, extracted from the [SDK v1.0.0rc601](https://github.com/frequenz-floss/frequenz-sdk-python/releases/tag/v1.0.0-rc601).
```
Comment (llucax) [inline RELEASE_NOTES.md:5]: I wouldn't remove this until we release v1.0.0, we can add that it is based on that code but with some updates.

File: RELEASE_NOTES.md [hunk 2]
```diff
@@ -2,8 +2,16 @@
...
+<!-- Here goes a general summary of what this release is about -->
+
+## Upgrading
+
+<!-- Here goes notes on how to upgrade from previous versions, including deprecations and what they should be replaced with -->
 
 ## New Features
 
-- Added support for `__round__` (`round(quantity)`), `__pos__` (`+quantity`) and `__mod__` (`quantity % quantity`) operators.
+- Add `QuantityT` type alias for `Quantity` and `QuantityLike` types.
```
Comment (llucax) [inline RELEASE_NOTES.md:13]: What is `QuantityLike`?

**Review by llucax — COMMENTED** [review]
I'm not sure if I would declare `QuantityT` here, I think this is only a requirement for the formula engine because of the time when `mypy` wasn't working and formula engine was added and the type checking was messed up. Ideally I think a `TypeVar` for quantities should be `TypeVar("QuantityT", bound=Quantity)`, so it resolves to the concrete type, I'm not sure why we had to go with the current version, I think this actually always will resolve to `Quantity`.

I think for now I would keep this as a hack in the SDK.

**Comment by llucax** [comment]
Just to clarify, I'm not completely against adding it here if it is better/more easy for some reason, just it wouldn't be my first choice for the reasons I mentioned above.

**Comment by ela-kotulska-frequenz** [comment]
As suggested QuantityT was moved back to frequenz-sdk

**Pull request closed without merging**